### PR TITLE
Support passing a From number to Twilio Messages API

### PR DESCRIPTION
**Pull request by aaronschachter**
* Adds optional `TWILIO_FROM_NUMBER` config variable to post a message using a [From number instead of a Messaging Service Sid](https://www.twilio.com/docs/api/rest/sending-messages#post-parameters-conditional). This allows us to post messages using our Mobile Commons Twilio API credentials by passing our shortcode or test longcode as the `From`.
    * Adds `/config/lib/twilio` to define Twilio config vars 

* Removes check for a `mobilecommons` query parameter, as we don't need an mData (#40) to post Mobile Commons messages to Conversations API per https://github.com/DoSomething/blink/pull/97

* Creates Conversation with medium `sms` instead of `twilio` for SMS Conversations

* `/receive-message` sets our Conversation medium as SMS by checking for expected Twilio parameters, so we're ready for Blink to sending messages received by our Mobile Commons test number: https://github.com/DoSomething/blink/issues/102
    * This essentially deprecates our Twilio route in Gambit Adapters, as Blink is posting directly to Conversations (not Adapters). I think this makes sense to do, since Conversations API will return an error if for whatever reason it can't process the message it receives.

### CODE REVIEW

File: lib/middleware/receive-message/params.js [hunk 1]
```diff
@@ -29,9 +17,12 @@ module.exports = function params() {
...
 
-    if (body.phone) {
-      req.platform = 'twilio';
-      req.userId = body.phone;
+    if (body.MessageSid) {
+      req.platform = 'sms';
+      req.userId = body.From;
+      req.inboundMessageText = body.Body;
+      // TODO: We need to save attachments here if we're ditching Gambit Adapters and Blink is
+      // posting directly to this /receive-message route.
```
Comment (aaronschachter) [inline lib/middleware/receive-message/params.js]: Seems it's time to decide to take or leave Gambit Adapters. 

If we want to keep it, Blink will need to POST to Adapters, which posts to Conversations, which posts to our medium (Twilio, Slack, etc). If there's an error in the Conversations API request, it needs to tell Adapters which tells Blink so it can retry. This feels like a lot of work for the sake of our Conversations API not handling how to parse an inbound message.

At the same time -- Twilio sends us a URL that redirects to an image, so we'll have to [port this code in Adapters over too](https://github.com/DoSomething/gambit-adapters/blob/562e8bc1085b4ee960ef86f9005802c713b791e9/app/routes/twilio.js#L25) into Receive Message middleware, to inspect and save attachments on inbound messages. 

I'm inclined to deprecate Adapters for the sake of less moving pieces. What do you think @rapala61 @sergii-tkachenko ?

File: config/lib/twilio.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+module.exports = {
+  accountSid: process.env.TWILIO_ACCOUNT_SID,
+  authToken: process.env.TWILIO_AUTH_TOKEN,
+  fromNumber: process.env.TWILIO_FROM_NUMBER,
```
Comment (aaronschachter) [inline config/lib/twilio.js]: We'll need to set this to our test longcode number on Conversations Staging to send replies to the received messages that Blink will be posting soon.

File: lib/middleware/receive-message/params.js (hunk 1 above)
Comment (aaronschachter) [inline lib/middleware/receive-message/params.js]: Refs #31

Comment (rapala61) [inline lib/middleware/receive-message/params.js]: I like the idea of creating a robust middleware that can massage the incoming POST requests to `receive-message` based on the medium/source. We could create another blink endpoint that 3rd party apps can POST to which then can be routed to the `receive-message`. We can add a query param to the webhook URL at the 3rd party app level to aid in parsing it `?medium=slack` for now. As we scale, and we start experiencing a lot more traffic, we can open a queue per service, but the consumers would POST to `receive-message`.

File: config/lib/twilio.js (hunk 2 above)
Comment (rapala61) [inline config/lib/twilio.js]: Is this one of those ENV variables that should explode the app if its not set? If so I think we should check their existence.

File: lib/middleware/receive-message/params.js (hunk 1 above)
Comment (aaronschachter) [inline lib/middleware/receive-message/params.js]: @rapala61 talked through this in Slack -- yes, let's have Blink post directly to our `/receive-message` endpoint, deprecating the use of Gambit Adapters for receiving Twilio messages.

We'll want to port [the code over from Adapters to save attachments in an inbound Twilio message.](https://github.com/DoSomething/gambit-adapters/blob/562e8bc1085b4ee960ef86f9005802c713b791e9/app/routes/twilio.js#L25)

File: config/lib/twilio.js (hunk 2 above)
Comment (aaronschachter) [inline config/lib/twilio.js]: It's questionable whether to explode the app... but probably makes sense. In order to deliver a reply message to a SMS Conversation, we need either the `TWILIO_FROM_NUMBER` or `TWILIO_MESSAGING_SERVICE_SID` config vars set.

An edge-case to make the argument for not exploding the app is if someone wanted to add Rivescript and test locally with consolebot -- we wouldn't be making any POST requests to Twilio in this case.

Comment (rapala61) [inline config/lib/twilio.js]: I'm not super familiar with Twilio message properties yet. Why do we need `TWILIO_FROM_NUMBER ` **OR** `TWILIO_MESSAGING_SERVICE_SID ` to reply back to an incoming message? I thought we always needed a From number since the SID was just an identifier of that particular message.

Comment (aaronschachter) [inline config/lib/twilio.js]: https://www.twilio.com/docs/api/rest/sending-messages#post-parameters-conditional

Our Twilio test code has been working without passing the From number thus far by passing a Messaging Service SID (created/maintainable through our Twilio console)

Comment (rapala61) [inline config/lib/twilio.js]: Ah, thanks. I confused the Account SID with the Message SID.
